flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/BufferManager.java

```diff
@@ -125,9 +125,9 @@ private boolean shouldContinueRequest(BufferPool bufferPool) {
 	}
 
 	/**
-	 * Requests exclusive buffers from the provider and returns the number of requested amount.
+	 * Requests exclusive buffers from the provider.
 	 */
-	int requestExclusiveBuffers() throws IOException {
+	void requestExclusiveBuffers() throws IOException {
 		Collection<MemorySegment> segments = globalPool.requestMemorySegments();
```

Comment:
The previous initial credit seems to be got from the returned collection size of `globalPool.requestMemorySegments()` somehow. If we can get the initial credit in advance, then it would be better to also pass it as argument into `requestMemorySegments` to get required size. To do so, the `NetworkBufferPool#requestMemorySegments` seems more general to decouple with credit semantic.